src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -1535,6 +1535,18 @@ public function testEnvCaseInsensitiveOnWindows()
         }
     }
 
+    public function testProcessWaitingCallback()
+    {
+        $p = $this->getProcessForCode('usleep(3000000);');
+        $callCounter = 0;
+        $p->wait(null,
+            function ($type, $buffer) use (&$callCounter) {
+                ++$callCounter;
+            }
+        );
+        $this->assertSame(2, $callCounter, 'The callback should be executed while waiting');
```

Comment:
Can you elaborate why you expect your callback to be called exactly twice?

Reply:
The executed process is a sleep for 3 seconds. And the Callback is called every second.